### DISCUSSION TITLE
Trying to understand why BonVision SphereMapping operator does not accept direct input

### DISCUSSION THREAD

**Discussion by banchan86** [bonsai-rx/bonsai#1835] [Q&A] [answered]
Following the example in the define display section of the basic workflow guide on https://bonvision.github.io/pages/01-Basic-Workflow/#draw-stimuli, it seems like spheremapping should accept any `DrawX` stimuli as input.

However, a direct connection does not work , but Publish/Subscribe subject pair with a single `DrawX` operator works (see attached bonsai workflow).

I'm just trying to understand why that is the case and if its a bug or simply something that I missed (as it seems that in the case of a single `DrawX` operator for the Publish/Subscribe subject pair it would pass on the same input as the direct connection). It would also help me in updating the BonVision documentation

[Spheremap-directinput-doesnotwork.zip](https://github.com/user-attachments/files/15681259/Spheremap-directinput-doesnotwork.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
`SphereMapping` operates on top of a given `ViewMatrix` (Matrix4 type) and renders all stored textures using/to a cubemap view/texture.

As a result, the input of `SphereMapping` must be some sort of `ViewMatrix` such that distortition from the mapping can be applied on top of it. 

If you go back to the example you sent (the one that doesnt work), the output of DrawCheckerboard is an `int` (if you are curious, it is actually the phase of the checkerboard :P).

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/9423d031-9b27-49df-8a9f-207fbcd46af5)

The secone example, the output of `OrthographicView` is the view matrix (`OpenTk.Matrix4`), so everything works fine.

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/a4b91ce0-9b2d-4a97-9341-9d3b090ae61d)

If you add the subject right after `OrthographicView` everything will still work since subjects behave as sinks.

Hope it helps!

**Reply by banchan86** [reply]
Thanks @bruno-f-cruz that was very helpful!
